### PR TITLE
Include all declared commodities in commodity groups and always have UC description

### LINKED ISSUE #98: DemoS(7–12) errors
> You can see the output of GAMS here:

Thanks.  I can see that the errors (for Demos7 at least) were not at all order issues, but quite some different kind of errors:  
1)  TRACO2 was not defined as a `COM_GRP` at all (domain). All commodities must be.
2)  `COM_STEP` parameters were written out without any commodity index
3)  The UC name 'UC_GrowthOfNewCars' was not defined as an `UC_N` at all (domain)

_Originally posted by @Antti-L in https://github.com/etsap-TIMES/times-excel-reader/issues/95#issuecomment-1658216131_

### PR DESCRIPTION
**Pull request by olejandro**
This PR adds commodities not explicitly specified in topology to commodity groups and uses user constrant name as user constraint description if user constraint description is missing.
Closes #98.

### CODE REVIEW

**Comment by Antti-L** [comment]
I could not quite find out how this fix is implemented, but anyway, all commodities defined in the `FI_Comm` tables should be included. I think they must be already collected from these tables because all commodities must have a type (NRG / MAT / DEM / ENV / FIN), and it is defined there. So, including all of them in `COM_GRP` should be trivial, and I wonder why some of the commodities were omitted earlier.

**Comment by olejandro** [comment]
Thanks @Antti-L. Well, as you can probably guess, based on the kind of commodities that were not included, previously the tool assumed that all the topology is defined explicitly and relied on it to derive `COM_GRP`. With this PR the info from `fi_comm` is also included. Please expect many more issues like this, before the first release. As always, really appreciate your feedback.

File: times_reader/transforms.py [hunk 1]
```diff
@@ -1253,7 +1253,7 @@ def name_comm_group(df):
             range="",
             filename="",
             uc_sets={},
-            tag="COMM_GROUPS",
+            tag="TOPOLOGY",
```
Comment (siddharth-krishna) [inline times_reader/transforms.py:1256]: If we've renamed the table to `TOPOLOGY`, would it make sense to also rename this function to `generate_topology`?

**Review by siddharth-krishna — APPROVED** [review]

**Comment by siddharth-krishna** [comment]
Quick note: this does not complete all the issues raised in #98, `PRC_ACTFLO` is still yet to be fixed, but since Demos 7-12 are now running through gams we could open a new issue for it perhaps?

File: times_reader/transforms.py (hunk 1 above)
Comment (olejandro) [reply times_reader/transforms.py:1256]: I thought about that. :-) However we already have `process_topology`. In `generate_commodity_groups` we are basically expanding `ProcessTopology`. I guess, we could depreciate one of them, but probably best to create a seperate PR focusing on this revision. I wonder whether we should wait until we are more clear on the `times_model` class?

**Comment by olejandro** [comment]
@siddharth-krishna it looks like you've already fixed the `PRC_ACTFLO` mapping in #115

**Comment by siddharth-krishna** [comment]
@olejandro I tried to but if you look at Demo 8 output it is still missing the ACTFLO table.. perhaps those rows got dropped somewhere earlier in the pipeline?

**Comment by olejandro** [comment]
Ah, I see. The mapping is correct now though, so we can deal with this as with all the other attributes for which output is missing. Let's see whether this gets alleviated once we include Veda defaults.

**Pull request merged**